tests/vxlan/test_vnet_route_leak.py

```diff
@@ -0,0 +1,180 @@
+import logging
+import pytest
+import re
+
+from tests.common.utilities import wait_until
+from vnet_constants import *
+from vnet_utils import cleanup_vnet_routes, cleanup_dut_vnets, cleanup_vxlan_tunnels, \
+                       apply_dut_config_files, generate_dut_config_files
+
+logger = logging.getLogger(__name__)
+
+pytestmark = [
+    pytest.mark.topology("t0")
+]
+
+BGP_WAIT_TIMEOUT = 120
+
+TESTING_STATUS = "Testing"
+CLEANUP_STATUS = "Cleanup"
+
+SHOW_VNET_ROUTES_CMD = "show vnet routes all"
+SHOW_BGP_SUMMARY_CMD = "show ip bgp summary"
+SHOW_BGP_ADV_ROUTES_CMD_TEMPLATE = "show ip bgp neighbor {} advertised-routes"
+RESTART_BGP_CMD = "sudo systemctl restart bgp"
+CONFIG_SAVE_CMD = "sudo config save -y"
+CONFIG_RELOAD_CMD = "sudo config reload -y"
+
+
+@pytest.fixture(scope="module")
+def configure_dut(minigraph_facts, duthost, vnet_config, vnet_test_params):
+    """
+    Setup/teardown fixture for VNET route leak test
+
+    During the setup portion, generates VNET VxLAN configurations and applies them to the DUT
+    During the teardown portion, removes all previously pushed VNET VxLAN information from the DUT
+
+    Args:
+        minigraph_facts: Minigraph information
+        duthost: DUT host object
+        vnet_config: Dictionary containing VNET configuration information
+        vnet_test_params: Dictionary containing VNET test parameters
+    """
+
+    duthost.shell("sonic-clear fdb all")
+    generate_dut_config_files(duthost, minigraph_facts, vnet_test_params, vnet_config)
+    apply_dut_config_files(duthost, vnet_test_params)
+
+    # In this case yield is used only to separate this fixture into setup and teardown portions
+    yield None
+
+    if vnet_test_params[CLEANUP_KEY]:
+        cleanup_vnet_routes(duthost, vnet_test_params)
+        cleanup_dut_vnets(duthost, minigraph_facts, vnet_config)
+        cleanup_vxlan_tunnels(duthost, vnet_test_params)
+
+        logger.info("Reloading CONFIG_DB from JSON")
+
+        duthost.shell(CONFIG_SAVE_CMD)
```

Comment:
Should this be `CONFIG_RELOAD`?

Reply:
And if that is the case can't we skip the `cleanup` steps?

Reply:
`CONFIG_SAVE` is correct as far as I know. The test itself involves saving some VNET information to `config_db.json` which shouldn't be there after the test is over, this `config save` is basically just to overwrite that.

Reply:
I would recommend backing up the original config DB and reloading it after the test is over, then. There is a lot of risk in running `config save` after a test case is complete b/c leftover configs can mess up later tests in confusing/hard to debug ways.

Reply:
Example backup: https://github.com/Azure/sonic-mgmt/blob/bdfb768e4ff3b9862642195ebf45d7c0d1aecbd3/tests/mvrf/test_mgmtvrf.py#L32

Example recovery: https://github.com/Azure/sonic-mgmt/blob/bdfb768e4ff3b9862642195ebf45d7c0d1aecbd3/tests/mvrf/test_mgmtvrf.py#L18

Reply:
Resolved offline. Will make a copy of `config_db.json` and restore that instead of relying on `config save`